internal/osutils/lockfile/pidlock.go

```diff
@@ -94,6 +96,28 @@ func (pl *PidLock) Close(keepFile ...bool) error {
 	return nil
 }
 
+// WaitForLock will attempt to acquire the lock for the duration given
+func (pl *PidLock) WaitForLock(timeout time.Duration) error {
+	lockedErr := &AlreadyLockedError{}
+	timer := time.NewTimer(timeout)
+	for {
+		_, err := pl.TryLock()
+		if err != nil {
+			if !errors.As(err, &lockedErr) {
+				return errs.Wrap(err, "Unexpected error attempting to acquire lock file")
```

Comment:
I think we should localize this and say something like "Could not write config as another process appears to be using it". Least that will give some indication to the user as to what might be the problem.

Reply:
I updated the error message here and also in the config package as I didn't want to mention the config file in this package.